src/main/java/org/jnode/fs/xfs/directory/BlockDirectoryEntry.java

```diff
@@ -111,12 +111,12 @@ public boolean isFreeTag() {
      * @return the offset size.
      */
     public long getOffsetSize() {
-        if (!isFreeTag) {
-            long l = 12 + nameSize - (isV5 ? 0 : 1);
+        if (!freeTag) {
+            long l = 12 + nameLength - (isV5 ? 0 : 1);
```

Comment:
Holy crap this code is annoying to read (yep I know it's not your code).

Reply:
It's to make the read operation always like 64-bit alignment.